### PR TITLE
KEYCLOAK-19388 correct AttributeConsumingService bug in SAML SP metadata

### PR DESCRIPTION
**Pull request by cgeorgilakis**
BUG FIX - version 15.0.0

### CODE REVIEW

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/KcSamlSpDescriptorTest.java [hunk 1]
```diff
@@ -41,43 +42,61 @@ public void testAttributeConsumingServiceIndexInSpMetadata() throws IOException,
...
             String spDescriptorString = identityProviderResource.export(null).readEntity(String.class);
             SAMLParser parser = SAMLParser.getInstance();
             EntityDescriptorType o = (EntityDescriptorType) parser.parse(new StringInputStream(spDescriptorString));
             SPSSODescriptorType spDescriptor = o.getChoiceType().get(0).getDescriptors().get(0).getSpDescriptor();
 
             assertThat(spDescriptor.getAttributeConsumingService().isEmpty(), is(false));
-            assertThat(spDescriptor.getAttributeConsumingService().get(0).getServiceName().get(0).getValue(), is("My Attribute Set"));
+            assertThat(spDescriptor.getAttributeConsumingService().get(0).getIndex(), is(12));
+            assertThat(spDescriptor.getAttributeConsumingService().get(0).getRequestedAttribute() != null, is(true));
+            assertThat(spDescriptor.getAttributeConsumingService().get(0).getRequestedAttribute().isEmpty(), is(false));
```
Comment (vramik) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/KcSamlSpDescriptorTest.java:77]:
```suggestion
            assertThat(spDescriptor.getAttributeConsumingService().get(0).getRequestedAttribute(), notNullValue());
            assertThat(spDescriptor.getAttributeConsumingService().get(0).getRequestedAttribute(), not(empty()));
```

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/KcSamlSpDescriptorTest.java [hunk 2]
```diff
@@ -41,43 +42,61 @@ public void testAttributeConsumingServiceIndexInSpMetadata() throws IOException,
...
+            assertThat(spDescriptor.getAttributeConsumingService().get(0).getRequestedAttribute().get(0).getName(), is("email_attr_name"));
+            assertThat(spDescriptor.getAttributeConsumingService().get(0).getRequestedAttribute().get(0).getFriendlyName(), is("email_attr_friendlyname"));
+            Assert.assertTrue(spDescriptor.getAttributeConsumingService().get(0).getServiceName() != null);
```
Comment (vramik) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/KcSamlSpDescriptorTest.java:80]:
```suggestion
            assertThat(spDescriptor.getAttributeConsumingService().get(0).getServiceName(), notNullValue());
```

**Review by vramik — CHANGES_REQUESTED** [review]
Thank you @cgeorgilakis for the PR, LGTM ... I just have few suggestions regarding the test. Please see inline, that same applies to the 3rd test as well.

**Comment by cgeorgilakis** [comment]
I did the requested changes in tests

**Review by vramik — APPROVED** [review]
Thank you @cgeorgilakis

**Comment by toby78** [comment]
Could anyone review this pull request please?

**Review by hmlnarik — APPROVED** [review]
Approving based on @vramik's review

**Comment by lscorcia** [comment]
Unfortunately this change has introduced hardcoded references to the user attribute mapper `"saml-user-attribute-idp-mapper"` , which broke https://github.com/keycloak/keycloak/pull/8312 .

I will rebase my pull request removing the hardcoded string with the appropriate detection of the `SamlMetadataDescriptorUpdater` interface.

**Pull request merged**